src/quo2/components/record_audio/record_audio/view.cljs

```diff
@@ -0,0 +1,473 @@
+                                (reanimated/animate-shared-value-with-delay width 56 83.33 :easing1 80)
+                                (reanimated/animate-shared-value-with-delay height 56 83.33 :easing1 80)
+                                (reanimated/animate-shared-value-with-delay border-radius-first-half 28 83.33 :easing1 80)
+                                (reanimated/animate-shared-value-with-delay border-radius-second-half 28 83.33 :easing1 80))
+           reset-x-animation #(do
+                                (reanimated/animate-shared-value-with-timing translate-x 0 100 :linear)
+                                (reanimated/set-shared-value connector-opacity 0)
+                                (reanimated/set-shared-value width 24)
+                                (reanimated/set-shared-value height 12)
+                                (reanimated/set-shared-value border-radius-first-half 8)
+                                (reanimated/set-shared-value border-radius-second-half 16))
+           fade-in-animation #(do
+                                (reanimated/animate-shared-value-with-timing translate-x 0 200 :linear)
+                                (reanimated/animate-shared-value-with-timing opacity 1 200 :linear))
+           fade-out-animation #(do
+                                 (reanimated/animate-shared-value-with-timing translate-x 20 200 :linear)
+                                 (reanimated/animate-shared-value-with-timing opacity 0 200 :linear)
+                                 (reanimated/set-shared-value connector-opacity 0)
+                                 (reanimated/set-shared-value width 24)
+                                 (reanimated/set-shared-value height 12)
+                                 (reanimated/set-shared-value border-radius-first-half 8)
+                                 (reanimated/set-shared-value border-radius-second-half 16))]
+       (effect! #(if @recording? (fade-in-animation) (fade-out-animation)) [@recording?])
+       (effect! #(cond @ready-to-delete? (start-x-animation) @recording? (reset-x-animation)) [@ready-to-delete?])
+       [:<>
+        [reanimated/view {:style (style/delete-button-container opacity)}
+         [reanimated/view {:style (style/delete-button-connector connector-opacity width height border-radius-first-half border-radius-second-half)}]]
+        [reanimated/view {:style          (style/delete-button translate-x opacity)
+                          :pointer-events :none}
+         [icons/icon :i/delete {:color colors/white
+                                :size  20}]]]))])
+
+(defn- record-button []
+  [:f>
+   (fn []
+     (let [opacity (reanimated/use-shared-value 1)
+           show-animation #(reanimated/set-shared-value opacity 1)
+           hide-animation #(reanimated/set-shared-value opacity 0)]
+       (effect! #(if (or @recording? @reviewing-audio?) (hide-animation) (show-animation)) [@recording? @reviewing-audio?])
```

Comment:
This line surely deserves some break lines ;)

Reply:
Done